.github/workflows/build.yml

```diff
@@ -28,9 +29,94 @@ jobs:
         terraform init -backend=false
         terraform validate -no-color
       working-directory: terraform
-
+  
     - name: Deploy to dev environment
       uses: ./.github/workflows/actions
       with:
         environment_name: dev
         azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+
+  deploy_test:
+    name: Deploy to test environment
+    needs: deploy_dev
+    runs-on: ubuntu-latest
+    environment: test
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 1.0.10
+
+    - name: Check formatting
+      run: terraform fmt -check
+      working-directory: terraform
+
+    - name: Validate
+      run: |
+        terraform init -backend=false
+        terraform validate -no-color
+      working-directory: terraform
+
+    - name: Deploy to test environment
+      uses: ./.github/workflows/actions
+      with:
+        environment_name: test
+        azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+
+  deploy_preprod:
+    name: Deploy to preprod environment
+    needs: deploy_test
+    runs-on: ubuntu-latest
+    environment: preprod
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 1.0.10
+
+    - name: Check formatting
+      run: terraform fmt -check
+      working-directory: terraform
+
+    - name: Validate
+      run: |
+        terraform init -backend=false
+        terraform validate -no-color
+      working-directory: terraform
+
+    - name: Deploy to preprod environment
+      uses: ./.github/workflows/actions
+      with:
+        environment_name: preprod
+        azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+
+  deploy_prod:
+    name: Deploy to prod environment
+    needs: deploy_preprod
+    runs-on: ubuntu-latest
+    environment: prod
+    # if: github.ref == 'refs/heads/main'
```

Comment:
Is this redundant now?

Reply:
Removed. There is environment check instead